src/backends/torch/torchlib.cc

```diff
@@ -254,6 +254,66 @@ namespace dd
     return out_val;
   }
 
+  c10::IValue TorchModule::extract(std::vector<c10::IValue> source,
+                                   std::string extract_layer)
+  {
+    if (_graph) // native modules take only one tensor as input for now
+      return _graph->extract(to_tensor_safe(source[0]), extract_layer);
+    if (_native)
+      return _native->extract(to_tensor_safe(source[0]), extract_layer);
+    auto output = _traced->forward(source);
+    if (output.isTensorList())
+      {
+        auto elems = output.toTensorList();
+        source = std::vector<c10::IValue>(elems.begin(), elems.end());
+      }
+    else if (output.isTuple())
+      {
+        auto &elems = output.toTuple()->elements();
+        source = std::vector<c10::IValue>(elems.begin(), elems.end());
+      }
+    else
+      {
+        source = { output };
+      }
+    c10::IValue out_val = source.at(_classif_in);
+    if (_hidden_states)
```

Comment:
Is this for LSTMs typically ?

Reply:
no, it allows to extract last layer from traced_net, ie for bert-like models

Reply:
I was initially added to TorchModule::forward to extract BERT hidden states and train a classification model with a linear head (_classif) on top of them. It is specific to BERT.

If I understand correctly, if there are no native and graph nets, `TorchModule::extract` returns the output of the traced model (`extract_layer` = "final").

EDIT: Maybe add a keyword for the specific extraction performed on BERT model?